CMakeLists.txt

```
@@ -1,73 +1,76 @@
-cmake_minimum_required(VERSION 3.22.1)
+cmake_minimum_required(VERSION 3.23)
 
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/CMake")
-include(ystdlib-cpp-helpers)
+project(ystdlib VERSION "0.1.0" LANGUAGES CXX)
 
-project(YSTDLIB_CPP LANGUAGES CXX)
-
-set(YSTDLIB_CPP_VERSION "0.0.1" CACHE STRING "Project version.")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
+include(ystdlib-helpers)
+include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
 
 option(BUILD_SHARED_LIBS "Build using shared libraries." OFF)
-option(YSTDLIB_CPP_BUILD_TESTING "Build the testing tree for ystdlib-cpp." ON)
+option(ystdlib_BUILD_TESTING "Build the testing tree for ystdlib." ON)
 
-# Require compiler versions that support the C++20 features necessary for compiling ystdlib-cpp
+# Require compiler versions that support the C++20 features necessary for compiling ystdlib
 if("AppleClang" STREQUAL "${CMAKE_CXX_COMPILER_ID}")
-    set(YSTDLIB_CPP_CMAKE_CXX_COMPILER_MIN_VERSION "16")
+    set(ystdlib_CMAKE_CXX_COMPILER_MIN_VERSION "16")
 elseif("Clang" STREQUAL "${CMAKE_CXX_COMPILER_ID}")
-    set(YSTDLIB_CPP_CMAKE_CXX_COMPILER_MIN_VERSION "16")
+    set(ystdlib_CMAKE_CXX_COMPILER_MIN_VERSION "16")
 elseif("GNU" STREQUAL "${CMAKE_CXX_COMPILER_ID}")
-    set(YSTDLIB_CPP_CMAKE_CXX_COMPILER_MIN_VERSION "11")
+    set(ystdlib_CMAKE_CXX_COMPILER_MIN_VERSION "11")
 else()
     message(
         FATAL_ERROR
         "Unsupported compiler: ${CMAKE_CXX_COMPILER_ID}. Please use AppleClang, Clang, or GNU."
     )
 endif()
-if("${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS "${YSTDLIB_CPP_CMAKE_CXX_COMPILER_MIN_VERSION}")
+if("${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS "${ystdlib_CMAKE_CXX_COMPILER_MIN_VERSION}")
     message(
         FATAL_ERROR
         "${CMAKE_CXX_COMPILER_ID} version ${CMAKE_CXX_COMPILER_VERSION} is too low. Must be at \
-        least ${YSTDLIB_CPP_CMAKE_CXX_COMPILER_MIN_VERSION}."
+        least ${ystdlib_CMAKE_CXX_COMPILER_MIN_VERSION}."
     )
 endif()
 
-# Enable exporting compile commands
 set(CMAKE_EXPORT_COMPILE_COMMANDS
     ON
     CACHE BOOL
     "Enable/Disable output of compile commands during generation."
     FORCE
 )
 
-if(YSTDLIB_CPP_IS_TOP_LEVEL)
+set(ystdlib_LIBRARIES
+    "containers"
+    "error_handling"
+    "io_interface"
+    "wrapped_facade_headers"
+    CACHE STRING
+    "Semi-colon separated list of libraries to be built."
+)
+
+message(STATUS "Building the following libraries:")
+foreach(LIB IN LISTS ystdlib_LIBRARIES)
+    message(" - ${LIB}")
```

```suggestion
    message(STATUS " - ${LIB}")
```

Unless we purposely didn't include `STATUS`?